api/restHandler/app/BuildPipelineRestHandler.go

```diff
@@ -304,6 +323,26 @@ func (handler PipelineConfigRestHandlerImpl) PatchCiMaterialSourceWithAppIdAndEn
 	common.WriteJsonResp(w, err, createResp, http.StatusOK)
 }
 
+func (handler PipelineConfigRestHandlerImpl) PatchCiMaterialSourceWithAppIdsAndEnvironmentId(w http.ResponseWriter, r *http.Request) {
+	bulkPatchRequest, userId, err := handler.parseBulkSourceChangeRequest(w, r)
+	if err != nil {
+		handler.Logger.Errorw("Parse error, PatchCiMaterialSource", "err", err, "PatchCiMaterialSource", bulkPatchRequest)
+		common.WriteJsonResp(w, err, nil, http.StatusBadRequest)
```

Comment:
for every error inside parseBulkSourceChangeRequest, we are already writing Json Resp. This will append two jsons and is not desired. Remove this.